bookkeeper-server/src/main/java/org/apache/bookkeeper/client/RackawareEnsemblePlacementPolicyImpl.java

```diff
@@ -79,7 +85,8 @@
     static final int REMOTE_MASK      = 0x04 << 24;
     static final int REMOTE_FAIL_MASK = 0x08 << 24;
     static final int READ_ONLY_MASK   = 0x10 << 24;
-    static final int UNAVAIL_MASK     = 0x20 << 24;
+    static final int SLOW_MASK        = 0x20 << 24;
+    static final int UNAVAIL_MASK     = 0x40 << 24;
```

Comment:
can't we preserve old value for constant ?

Reply:
No, the order is important, since candidate bookies are sorted by these masks. You want to select a slow bookie before an unavailable one. You probably want to select a slow bookie before a readonly bookie too.

Reply:
Ok. I understand